crates/types/src/config/networking.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::retries::RetryPolicy;
+
+/// # Networking options
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, derive_builder::Builder)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "schemars", schemars(rename = "WorkerOptions", default))]
+#[serde(rename_all = "kebab-case")]
+#[builder(default)]
+pub struct NetworkingOptions {
+    /// # Retry policy
+    ///
+    /// Retry policy to use for internal node-to-node networking.
+    pub retry_policy: RetryPolicy,
+}
+
+impl NetworkingOptions {}
+
+const NETWORKING_DEFAULT_RETRY_INITIAL_INTERVAL: Duration = Duration::from_millis(10);
+const NETWORKING_DEFAULT_RETRY_MAX_INTERVAL: Duration = Duration::from_millis(500);
+const NETWORKING_DEFAULT_RETRY_MAX_ATTEMPTS: usize = 10;
+
+impl Default for NetworkingOptions {
+    fn default() -> Self {
+        Self {
+            retry_policy: RetryPolicy::exponential(
+                NETWORKING_DEFAULT_RETRY_INITIAL_INTERVAL,
+                2.0,
+                Some(NETWORKING_DEFAULT_RETRY_MAX_ATTEMPTS),
+                Some(NETWORKING_DEFAULT_RETRY_MAX_INTERVAL),
+            ),
```

Comment:
What's the duration before we give up? ~3s?

Reply:
Correct, depending on jitter it's on the order of about 3s, up from the previous (10 * 250ms) * jitter.

Sample run:

```
Attempt 1/Some(10): 13ms
Attempt 2/Some(10): 24.737462ms
Attempt 3/Some(10): 50.628101ms
Attempt 4/Some(10): 98.166179ms
Attempt 5/Some(10): 187.562579ms
Attempt 6/Some(10): 401.586045ms
Attempt 7/Some(10): 644.410285ms
Attempt 8/Some(10): 507.701047ms
Attempt 9/Some(10): 590.740606ms
Attempt 10/Some(10): 629.268748ms
Total: 3.147801052s
```